radix-engine-interface/src/blueprints/component.rs

```diff
@@ -0,0 +1,157 @@
+use core::marker::PhantomData;
+
+use radix_engine_common::prelude::*;
+
+pub trait TypeInfoMarker {
```

Comment:
I wonder if we should just move the following trait from `scrypto/component` to here?
```
pub trait HasTypeInfo {
    const PACKAGE_ADDRESS: Option<PackageAddress>;
    const BLUEPRINT_NAME: &'static str;
    const OWNED_TYPE_NAME: &'static str;
    const GLOBAL_TYPE_NAME: &'static str;

    fn blueprint_id() -> BlueprintId {
        let package_address = Self::PACKAGE_ADDRESS.unwrap_or(Runtime::package_address());
        BlueprintId::new(&package_address, Self::BLUEPRINT_NAME)
    }
}
```

And then it could also implement eg `get_own_type_data` / `get_reference_type_data` to also commonise that between engine and scrypto?

Reply:
I agree, will make that change.

Reply:
Explored this a little bit.

In `HasTypeInfo`, the `PACKAGE_ADDRESS` is optional. Thus, in cases when the package address is `None`, the `BlueprintId` is obtained through `Runtime::package_address()` which is of course not available in the engine. So, I'd rather keep them separate for the time being.